### PR TITLE
Set pause annotation on MC apps

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/14993

This is an implementation bypassing admissionkit related changes - in case I don't have enough time this week.

### CODE REVIEW

**Comment by kopiczko** [comment]
Tested locally on 1.18 cluster.

**Review by rossf7 — APPROVED** [review]

**Review by kubasobon — COMMENTED** [review]
Since `MutatingWebhookConfiguration` is attached to both `CREATE` and `UPDATE` operations (https://github.com/giantswarm/app-admission-controller/blob/master/helm/app-admission-controller/templates/webhook.yaml#L26-L28), we will never get rid of pause annotation. Every update will bring it back.
How about a check: If config version annotation is set, do not add a pause annotation.

**Comment by kopiczko** [comment]
That's a good call.

**Comment by kopiczko** [comment]
I think we can simply check existence of creation timestamp

**Comment by rossf7** [comment]
Yes nice catch.

@kopiczko Checking the creation timestamp will work but the webhook request includes the type of operation. So that is another option. https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#request

**Comment by kopiczko** [comment]
> think we can simply check existence of creation timestamp

I tested that and it isn't good enough. If you have `creationTimestamp: "..."` set in the manifest during creation it will not create the annotation. I'll need to make that based on the even type as @rossf7 suggests.

File: pkg/app/mutate_app.go [hunk 1]
```diff
@@ -77,12 +78,12 @@ func (m *Mutator) Mutate(request *v1beta1.AdmissionRequest) ([]mutator.PatchOper
 
 	// We check the deletion timestamp because app CRs may be deleted by
 	// deleting the namespace they belong to.
-	if !appNewCR.DeletionTimestamp.IsZero() {
+	if request.Operation == v1beta1.Update && !appNewCR.DeletionTimestamp.IsZero() {
```
Comment (kopiczko) [inline pkg/app/mutate_app.go:81]: I made this check explicit so there to hopefully reduce confusion.

Comment (rossf7) [reply pkg/app/mutate_app.go:81]: That's a good shout, thanks.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**